tools/CMakeLists.txt

```diff
@@ -19,9 +19,6 @@ endif()
 
 set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
 
-# checkout submodules
-include(${SOF_ROOT_SOURCE_DIRECTORY}/scripts/cmake/git-submodules.cmake)
```

Comment:
If rimage is not brought with SOF, from the start (something that is not happening right now with Zephyr - see https://buildkite.com/zephyr/zephyr/builds/37293#3ee62b3f-1854-4979-9e89-9497ee14f66e) and later is trying to bring it using ["Git submodules update"](https://buildkite.com/zephyr/zephyr/builds/37293#3ee62b3f-1854-4979-9e89-9497ee14f66e/509-610) from `git-submodules.cmake`. If this is now removed how rimage will be added to SOF fork from Zephyr?

Reply:
Zephyr CI does not need rimage right now, not before I added the Zephyr -> smex -> rimage dependency in commit f6c71c21d03b8111 which is not in Zephyr yet (and that you are adding via https://github.com/zephyrproject-rtos/zephyr/pull/37250 )

The Zephyr -> smex -> rimage dependency should be addressed (will look at it very soon) but it should not be through the totally unrelated `tools/CMakeLists.txt`. Hence this cleanup PR.

In fact running `git submodule update` in CMake is a questionable convenience hack already, even when performed in the right CMake place. Cloning and building should be separate steps and they're almost always are.

The SOF -> smex -> rimage dependency is already addressed.

Reply:
@marc-hb thanks for the explanation. It's clear now. 

I know that rimage is used for signing (with` west sign`).
How will rimage be brought to SOF with Zephyr?
For example, me, locally I clone it separately and with `west sign` I give the absolute path to my cloned rimage.

The SOF signing is not tested with Zephyr CI?
I'm asking this because even if we apply these patches on SOF and also on SOF fork from zephyrproject, without rimage submodule we will still have a failed buildkite or not?

Reply:
Zephyr CI does not build or use rimage right now. Only SOF CI does, with this script: https://github.com/thesofproject/sof/blob/c73ec9f981d5f7b4a/scripts/xtensa-build-zephyr.sh

> I'm asking this because even if we apply these patches on SOF and also on SOF fork from zephyrproject, without rimage submodule we will still have a failed buildkite or not?

If you take a closer look at the buildkite failure you will see two spurious rimage dependencies and two rimage build failures unrelated to each other: one with `tools/sof-logger` and another one with `smex`. See my last comment in https://github.com/zephyrproject-rtos/zephyr/pull/37250

This PR fixes only the sof-logger failure (please confirm), I hope to submit a fix for the smex failure today. Let's discuss in https://github.com/zephyrproject-rtos/zephyr/pull/37250 if needed.